dev/release/src/config.ts

```diff
@@ -34,63 +40,213 @@ export interface Config {
+    writeFileSync(releaseConfigPath, JSON.stringify(config, null, 2))
+}
+
+export function newRelease(
+    version: SemVer,
+    releaseDate: DateTime,
+    captainGithub: string,
+    captainSlack: string
+): ScheduledReleaseDefinition {
+    return {
+        ...releaseDates(releaseDate),
+        current: version.version,
+        captainGitHubUsername: captainGithub,
+        captainSlackUsername: captainSlack,
     }
-    const parsedUpcoming = semver.parse(config.upcomingRelease, parseOptions)
-    if (!parsedUpcoming) {
-        throw new Error(`config.upcomingRelease '${config.upcomingRelease}' is not valid semver`)
+}
+
+export async function newReleaseFromInput(versionOverride?: SemVer): Promise<ScheduledReleaseDefinition> {
+    let version = versionOverride
+    if (!version) {
+        version = await selectVersionWithSuggestion('Enter the desired version number')
     }
 
-    // Verify the configured upcoming release. The response is cached and expires in a
-    // week, after which the captain is required to confirm again.
-    const now = new Date()
-    const cachedVersionResponse = `${cacheFolder}/current_release_${now.getUTCFullYear()}_${getWeekNumber(now)}.txt`
-    const confirmVersion = await readLine(
-        `Please confirm the upcoming release version configured in '${configPath}' (currently '${config.upcomingRelease}') by entering it again: `,
-        cachedVersionResponse
+    const releaseDateStr = await retryInput(
+        'Enter the release date (YYYY-MM-DD). Enter blank to use current date: ',
+        val => {
+            if (val && /^\d{4}-\d{2}-\d{2}$/.test(val)) {
+                return true
+            }
+            // this will return false if the input doesn't match the regexp above but does exist, allowing blank input to still be valid
+            return !val
+        },
+        'invalid date, expected format YYYY-MM-DD'
     )
-    const parsedConfirmed = semver.parse(confirmVersion, parseOptions)
-    let error = ''
-    if (!parsedConfirmed) {
-        error = `Provided version '${confirmVersion}' is not valid semver`
-    } else if (semver.neq(parsedConfirmed, parsedUpcoming)) {
-        error = `Provided version '${confirmVersion}' and config.upcomingRelease '${config.upcomingRelease}' do not match - please update the release configuration at '${configPath}' and try again`
+    let releaseTime: DateTime
+    if (!releaseDateStr) {
+        releaseTime = DateTime.now().setZone('America/Los_Angeles')
+        console.log(chalk.blue(`Using current time: ${releaseTime.toString()}`))
+    } else {
+        releaseTime = DateTime.fromISO(releaseDateStr, { zone: 'America/Los_Angeles' })
     }
 
-    // If error, abort and remove the cached response (since it is invalid anyway)
-    if (error !== '') {
-        unlinkSync(cachedVersionResponse)
-        throw new Error(error)
+    const captainGithubUsername = await retryInput('Enter the github username of the release captain: ', val => !!val)
```

Comment:
```suggestion
    const captainGithubUsername = await retryInput('Enter the github username of the release captain: ', val => Boolean(val))
```

This seems easier for anyone to understand IMO.

Reply:
Got it. I generally find it hard to understand lol, that's why I most times opt for `Boolean` over that but since there's no laid out convention for this, I'm fine either way.